File: packages/zynth-auth-session/android/ZynthAuthSession/src/main/java/dev/zynth/authsession/ZynthAuthSession.kt
package dev.zynth.authsession

import androidx.activity.ComponentActivity
import com.zynth.kit.runtime.ZynthRuntime

object ZynthAuthSession {
  @JvmStatic
  fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
    val module = AuthSessionModule(activity, runtime)
    runtime.installModules(listOf(module))
  }
}

File: packages/zynth-auth-session/ios/ZynthAuthSession.swift
import Foundation
import ZynthKit

@objc(ZynthAuthSession)
public final class ZynthAuthSession: NSObject {
  private static var module: AuthSessionModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard module == nil else {
      return
    }

    let instance = AuthSessionModule(runtime: runtime)
    runtime.installModules([instance])
    module = instance
  }

  @objc public static func cleanup() {
    module = nil
  }
}
